spot_driver/include/spot_driver/robot_state/spot_robot_state_middleware_handle.hpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#pragma once
+
+#include <rclcpp/node.hpp>
+#include <spot_driver/interfaces/rclcpp_logger_interface.hpp>
+#include <spot_driver/interfaces/rclcpp_parameter_interface.hpp>
+#include <spot_driver/interfaces/rclcpp_tf_interface.hpp>
+#include <spot_driver/interfaces/rclcpp_wall_timer_interface.hpp>
+#include <spot_driver/robot_state/spot_robot_state_publisher.hpp>
+#include <spot_driver/types.hpp>
+#include <tl_expected/expected.hpp>
+
+#include <memory>
+
+namespace spot_ros2 {
+
+/**
+ * @brief Production implementation of a SpotRobotStatePublisher::MiddlewareHandle
+ */
+
+class SpotRobotStateMiddlewareHandle : public SpotRobotStatePublisher::MiddlewareHandle {
```

Comment:
Do we really need all these classes to start with the prefix "SpotRobot"?

Reply:
I renamed all these to be `StatePublisher`, etc.

There was a potential for confusion with ROS's RobotStatePublisher, so if this is just the StatePublisher in the spot_driver package then hopefully it'll be clear enough.